include/boost/gil/gil_concept.hpp

```diff
@@ -33,6 +33,9 @@
 
 namespace boost { namespace gil {
 
+#pragma GCC diagnostic push
```

Comment:
This should be GCC-specific, otherwise "unknown pragma" will be issued eg. by MSVC

```
#if defined(__GNUC__) && (__GNUC__ >= 4)
#pragma GCC diagnostic ignored "-Wunused-local-typedefs"
#endif
```